src/lib/functions/utils.js

```diff
@@ -21,8 +19,40 @@ const detectAwsSdkError = ({ error }) => {
 
 const formatLambdaError = (err) => chalk.red(`${err.errorType}: ${err.errorMessage}`)
 
+const exceptionsList = new Set([
+  'application/csp-report',
+  'application/graphql',
+  'application/json',
+  'application/javascript',
+  'application/x-www-form-urlencoded',
+  'application/x-ndjson',
+  'application/xml',
+])
+
+/**
+ * @param {string | undefined} contentType
+ * @returns {boolean}
+ */
 const shouldBase64Encode = function (contentType) {
-  return Boolean(contentType) && BASE_64_MIME_REGEXP.test(contentType)
+  if (!contentType) {
+    return true
+  }
+
+  contentType = contentType.toLowerCase()
+
+  if (contentType.startsWith('text/')) {
```

Comment:
Any reason why we can't move these to a single condition?

Reply:
I like how each of the conditions is its own case, à la "if it's a text, don't encode; if it's another content-type encoded as json or xml, don't encode; here's a set of exceptions we don't encode". Moving them into a single condition would make that grouping harder to see. Can you elaborate on why we should move them into a single condition?

Reply:
Personally, I think having 3 termination points when you can have 1 makes it harder to trace and debug. I also don't see how having a single condition makes the grouping harder to see. It's not a big deal though, and I respect the personal preference if you think it adds more clarity.